File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:

        # create the empty dictonary
        freq = dict()
        for i, num in enumerate(nums):
            if num in freq:
                return True
            freq[num] = freq.get(num,0)+1

        return False

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
         HashMap<Integer,Integer> freq = new HashMap<>();

        for(int num: nums){
            freq.put(num,freq.getOrDefault(num,0)+1);
        }

        Queue<Integer> minHeap = new PriorityQueue<>(
            (a,b) -> freq.get(a) - freq.get(b)
        );

        for(int i: freq.keySet()){

            minHeap.add(i);
            if(minHeap.size()>k){
                minHeap.poll();
            }

        }

       // List<Integer> topElement = new ArrayList<>();
        int[] topElement = new int[k];

        for(int i= 0; i < k; i++){
            topElement[i]= minHeap.poll();
           
        }

        
        return topElement;
        
    }
}
